ddev/src/ddev/e2e/agent/docker.py

```diff
@@ -282,6 +282,12 @@ def invoke(self, args: list[str]) -> None:
     def enter_shell(self) -> None:
         self._run_command(self._format_command(['cmd' if self._is_windows_container else 'bash']), check=True)
 
+    @retry(wait=wait_fixed(2), stop=stop_after_attempt(3))
```

Comment:
Can the retry be made conditional based on https://www.dockerstatus.com/? If the site is down, then retries won't help. Or do we already know that docker is guaranteed to be up while pulling the image?

Also maybe include the docker status result in the exception message?

Reply:
That wouldn't be worth the effort and also the status wouldn't be tied to everything that could fail locally.

Reply:
> If the site is down, then retries won't help.

That's true, but in that case everything will fail and retrying won't hurt either. 

I agree with ofek here, I don't think this is worth the effort at least for now. This is the first time I see this error